ingest/ingest-datasets-old.py

```diff
@@ -0,0 +1,392 @@
+
+    # portal_groups
+    portal_groups = get_portal_groups(ds)
+    if portal_groups:
+        doc.update({"portalGroups": portal_groups})
+
+    # projects NOT WORKING YET
+    projects = get_projects(ds)
+    if projects:
+        doc.update({"projects": projects})
+
+    # dataType
+    data_types = get_data_types(ds)
+    if data_types:
+        doc.update({"dataTypes": data_types})
+
+    # cites NOT WORKING YET
+    cites = get_cites(ds)
+    if cites:
+        doc.update({"citations": cites})
+
+    # authors
+    authors = []
+    authorships = [faux for faux in ds.objects(VIVO.relatedBy) if has_type(faux, VIVO.Authorship)]
+    for authorship in authorships:
+
+        author = [person for person in authorship.objects(VIVO.relates) if has_type(person, FOAF.Person)][0]
+        name = author.label().toPython() if author else None
+
+        obj = {"uri": str(author.identifier), "name": name}
+
+        rank = list(authorship.objects(VIVO.rank))
+        rank = str(rank[0].toPython()) if rank else None # added the str()
+        if rank:
+            obj.update({"rank": rank})
+
+        research_areas = [research_area.label().toPython() for research_area in author.objects(VIVO.hasResearchArea) if research_area.label()]
+
+        if research_areas:
+            obj.update({"researchArea": research_areas})
+
+        authors.append(obj)
+
+        org = list(author.objects(DCO.inOrganization))
+        org = org[0] if org else None
+        if org and org.label():
+            obj.update({"organization": {"uri": str(org.identifier), "name": org.label().toPython()}})
+
+    try:
+        authors = sorted(authors, key=lambda a: a["rank"]) if len(authors) > 1 else authors
+    except KeyError:
+        print("missing rank for one or more authors of:", dataset)
+
+    doc.update({"authors": authors})
+
+    # distributions
+    distributions = []
+    distributionList = [faux for faux in ds.objects(DCO.hasDistribution) if has_type(faux, DCAT.Distribution)]
+    for distribution in distributionList:
+        accessURL = str(list(distribution.objects(DCO.accessURL))[0])
+        name = distribution.label().toPython() if distribution else None
+        obj = {"uri": str(distribution.identifier), "accessURL": accessURL, "name": name}
+
+        fileList = list(distribution.objects(DCO.hasFile))
+        # fileList = [f for f in distribution.objects(DCO.hasFile) if has_type(f, DCO.File)]
+        fileList = fileList if fileList else None
+        files = []
+        for file in fileList:
+            downloadURL = list(file.objects(DCO.downloadURL))
+            downloadURL = str(downloadURL[0]) if downloadURL else None
+            fileObj = {"uri": str(file.identifier),
+                       "name": file.label().toPython()}
+            fileObj.update({"downloadURL": downloadURL})
+            files.append(fileObj)
+
+        if files:
+            obj.update({"files": files})
+
+        distributions.append(obj)
+
+    # try:
```

Comment:
Looks like authors is taken care of above so let's remove this commented out code.